src/main/java/uk/gov/hmcts/ccd/domain/service/createevent/AuthorisedCreateEventOperation.java

```diff
@@ -71,12 +64,15 @@ public CaseDetails createCaseEvent(String uid,
             throw new ValidationException("Cannot find user roles for the user");
         }
 
+        String caseTypeId = existingCaseDetails.getCaseTypeId();
+        final CaseType caseType = caseDefinitionRepository.getCaseType(caseTypeId);
+        if (caseType == null) {
```

Comment:
is this null check required? I think the repository already throws ResournceNotFoundException so this check is redundant.

Reply:
There is no null check in the repository. In fact all the other users do the null check at the moment.